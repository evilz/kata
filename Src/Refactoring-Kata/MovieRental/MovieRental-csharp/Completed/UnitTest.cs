using System;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace MovieRental_csharp.Completed
{
    public class UnitTest
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void TestStatement()
        {
            Customer customer = new Customer("jpartogi");

            customer.AddNewRental("REGULAR3", Movie.REGULAR, 3);
            customer.AddNewRental("REGULAR1", Movie.REGULAR, 1);
            customer.AddNewRental("NEW_RELEASE3", Movie.NEW_RELEASE, 3);
            customer.AddNewRental("NEW_RELEASE1", Movie.NEW_RELEASE, 1);
            customer.AddNewRental("CHILDRENS1", Movie.CHILDRENS, 1);
            customer.AddNewRental("CHILDRENS5", Movie.CHILDRENS, 5);
            
            var statement = customer.Statement();
            ApprovalTests.Approvals.Verify(statement);
        }


    }

    public static class TestHelpers
    {
        public static void AddNewRental(this Customer customer, string title, int priceCode, int daysRented)
        {
            var movie = new Movie(title, priceCode);
            var rental = new Rental(movie, daysRented);
            customer.AddRental(rental);
        }
    }
}
