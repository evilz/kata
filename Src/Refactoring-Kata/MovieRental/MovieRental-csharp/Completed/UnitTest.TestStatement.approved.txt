Rental Record for jpartogi
	REGULAR3	3,5
	REGULAR1	2
	NEW_RELEASE3	9
	NEW_RELEASE1	3
	CHILDRENS1	1,5
	CHILDRENS5	4,5
Amount owed is 23,5
You earned 7 frequent renter points